src/lib.rs

```diff
@@ -39,6 +39,39 @@ pub fn parse_and_inline_modules(src_file: &Path) -> syn::File {
         .output
 }
 
+/// The string `syn_inline_mod_path`. This acts as a marker for inlined paths.
+pub static SYN_INLINE_MOD_PATH: &str = "syn_inline_mod_path";
+
+/// Given a list of attributes from an inlined module, search for the first attribute of the form
+/// `#[syn_inline_mod_path(b"...")]`. If it is found, then return a triple with:
+/// * the module path
+/// * the outer attributes
+/// * the inner attributes
+///
+/// See the documentation for `InlinerBuilder::annotate_paths` for more information.
+pub fn find_mod_path(attrs: &[Attribute]) -> Option<(PathBuf, &[Attribute], &[Attribute])> {
```

Comment:
A tuple type with two members of the same data type isn't great for maintainability/readability. How would you feel about making a struct that has these methods:

```rust
pub struct ModAttributes<'a> {
    // fields elided
}

impl ModAttributes<'_> {
    pub fn path(&self) -> Option<PathBuf> {
        unimplemented!()
    }

    pub fn outer_attributes(&self) -> &[Attribute] {
        unimplemented!()
    }

    pub fn inner_attributes(&self) -> &[Attribute] {
        unimplemented!()
    }
}
```

That would also provide a natural place to put a comment explaining that inner attributes from the inlined file haven't been converted to outer attributes on the new module.

Aside: I briefly entertained the notion of this being a trait rather than a struct, but 1) this way the struct can store the index of the path attribute to avoid recalculating it for each of the accessor methods and 2) I think having a trait that's just implemented for `&[Attribute]` is a little too hard to find.

Reply:
Yeah I agree -- pretty sure I was just being lazy here :) Will do.